modules/pricer/src/main/java/com/opengamma/strata/pricer/capfloor/VolatilityIborCapFloorLegPricer.java

```diff
@@ -227,6 +230,44 @@ public CurrencyAmount currentCash(
         .orElse(CurrencyAmount.zero(capFloorLeg.getCurrency()));
   }
 
+  //-------------------------------------------------------------------------
+  /**
+   * Calculates the forward rates for each caplet/floorlet of the Ibor cap/floor leg.
+   *
+   * @param capFloorLeg  the Ibor cap/floor leg
+   * @param ratesProvider  the rates provider
+   * @return the forward rates
+   */
+  public Map<IborCapletFloorletPeriod, Double> forwardRates(
```

Comment:
Maybe there should be a bean to represent this map?

Reply:
Would you recommend two beans, `IborCapletFloorletPeriodForwardRates` and `IborCapletFloorletPeriodVolatilities`, or just one bean that handles both, as they both just a map of period to double?

Reply:
We tend to share classes in results, eg `CurrencyAmount`. So I think `IborCapletFloorletPeriodAmounts`